### ISSUE TITLE
a new release with redis 4.3 compatibility

### ISSUE THREAD

**Issue by lennart** [splitrb/split#647]
I was wondering whether there could be a new release of the gem that avoids these deprecation warnings with redis >= 4.2:

```
Redis#exists(key)` will return an Integer in redis-rb 4.3. `exists?` returns a boolean, you should use it instead.
To opt-in to the new behavior now you can set Redis.exists_returns_integer =  true.
To disable this message and keep the current (boolean) behaviour of 'exists' you can set `Redis.exists_returns_integer = false`, but this option will be removed in 5.0. (/home/m/.anyenv/envs/rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/split-3.4.1/lib/split/experiment_catalog.rb:16:in `find')
```

As far as I understand, current master branch of split is already compatible, but the released version 3.4.1 is not.

It would be great to have a released version as other projects (e.g. sidekiq) already require redis >= 4.2 so users of both will be locked to older versions.

Thanks for your effort!

**Comment by andrehjr** [comment]
Thanks for the issue @lennart! I'm planning to release 4.0.0.rc1 soonish as we dropped support for some older versions.

**Comment by lennart** [comment]
great, thanks to hear!

**Comment by andrehjr** [comment]
Sorry for the late response. I've just released v4.0.0pre. Can you try that? :)
